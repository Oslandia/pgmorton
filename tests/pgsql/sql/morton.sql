CREATE EXTENSION postgis;
CREATE EXTENSION morton;

SELECT Morton_Script_Version();

SELECT Morton_Encode(-1, 2);
SELECT Morton_Encode(6, 7);
SELECT Morton_Encode(8.7, 10.7, Morton_GridMake(3.5, 2.5, 12.7, 14.9, 4));

SELECT Morton_Decode(62);
SELECT Morton_Decode(morton_encode(6, 7));

SELECT Morton_Revert(62);
SELECT Morton_Revert(morton_encode(6, 7));

SELECT Morton_GridMake(1.2, 3.5, 63.2, 45.2, 64);
SELECT Morton_GridMake(728262.803178, 4676439.353454, 728998.135178, 4677014.685454, 64);
SELECT Morton_GridMake(Box2d(ST_GeomFromText('LINESTRING(3.5 2.5, 12.7 11.8)')), 4);

SELECT Morton_GridAsText(Morton_GridMake(Box2d(ST_GeomFromText('LINESTRING(3.5 2.5, 12.7 11.8)')), 4));

SELECT Morton_Gridded(8.7, 10.7, Morton_GridMake(3.5, 2.5, 12.7, 14.9, 4));
