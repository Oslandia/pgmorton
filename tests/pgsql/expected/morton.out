CREATE EXTENSION postgis;
CREATE EXTENSION morton;
SELECT Morton_Script_Version();
 morton_script_version 
-----------------------
 0.0.1
(1 row)

SELECT Morton_Encode(-1, 2);
ERROR:  coordinates are negatives
SELECT Morton_Encode(6, 7);
 morton_encode 
---------------
            62
(1 row)

SELECT Morton_Encode(8.7, 10.7, Morton_GridMake(3.5, 2.5, 12.7, 14.9, 4));
 morton_encode 
---------------
            12
(1 row)

SELECT Morton_Decode(62);
 morton_decode 
---------------
 {6,7}
(1 row)

SELECT Morton_Decode(morton_encode(6, 7));
 morton_decode 
---------------
 {6,7}
(1 row)

SELECT Morton_Revert(62);
 morton_revert 
---------------
    2080374784
(1 row)

SELECT Morton_Revert(morton_encode(6, 7));
 morton_revert 
---------------
    2080374784
(1 row)

SELECT Morton_GridMake(1.2, 3.5, 63.2, 45.2, 64);
                morton_gridmake                 
------------------------------------------------
 ( 1.200000, 3.500000, 0.968750, 0.651563, 64 )
(1 row)

SELECT Morton_GridMake(728262.803178, 4676439.353454, 728998.135178, 4677014.685454, 64);
                      morton_gridmake                       
------------------------------------------------------------
 ( 728262.803178, 4676439.353454, 11.489562, 8.989562, 64 )
(1 row)

SELECT Morton_GridMake(Box2d(ST_GeomFromText('LINESTRING(3.5 2.5, 12.7 11.8)')), 4);
                morton_gridmake                
-----------------------------------------------
 ( 3.500000, 2.500000, 2.300000, 2.325000, 4 )
(1 row)

SELECT Morton_GridAsText(Morton_GridMake(Box2d(ST_GeomFromText('LINESTRING(3.5 2.5, 12.7 11.8)')), 4));
                                           morton_gridastext                                            
--------------------------------------------------------------------------------------------------------
 {"offsetx": 3.500000, "offsety": 2.500000, "cell_width": 2.300000, "cell_height": 2.325000, "size": 4}
(1 row)

SELECT Morton_Gridded(8.7, 10.7, Morton_GridMake(3.5, 2.5, 12.7, 14.9, 4));
 morton_gridded 
----------------
 {2,2}
(1 row)

